### ISSUE TITLE
The listbox scrollbar behaves abnormally

### ISSUE THREAD

**Issue by NeverMorewd** [AvaloniaUI/Avalonia#17349]
### Describe the bug

After adding several items to ListBox, scroll the listbox with draging ScrollBar or code: `listbox.AutoScrollToSelectedItem = true;listbox.SelectedItem = item`  will raise an abnormal display.

![image](https://github.com/user-attachments/assets/2da8b622-54cf-446e-97ac-672f4f4104a6)

### To Reproduce

clone: https://github.com/NeverMorewd/Lemon.ModuleNavigation
start: Lemon.ModuleNavigation.Sample.DesktopHosting
Click ModuleB and ModuleC alternately to add enough items to ListBox for showing ScrollBar
Drag ScrollBar slowly up and down will produce this issue

![20241024223412_rec_](https://github.com/user-attachments/assets/f6f1ce0e-80e5-403c-acca-383f8a97690d)

### Avalonia version

11.1.3

### OS

Windows

**Comment by timunie** [comment]
Can you please provide a minimum sample to reproduce? We have no capacity to debug into a more complex sample.

**Comment by NeverMorewd** [comment]
> Can you please provide a minimum sample to reproduce? We have no capacity to debug into a more complex sample.

up to now, I have not reproduced with a mini project
And I found two old issues which seems to be related with this issue.
https://github.com/AvaloniaUI/Avalonia/issues/10585
https://github.com/AvaloniaUI/Avalonia/issues/10367

In addition
If i commented out this line, the listboxitem overlap issue will not produce.
![image](https://github.com/user-attachments/assets/b99f0f06-bf9e-412d-af71-0d645c3b04d5)

**Comment by NeverMorewd** [comment]
The overlapping ListBoxItem issue can now be avoided using ScrollToView, but the scrollbar's abnormal behavior still persists. When you hover over the scrollbar, hold down the left mouse button and drag up/down, the mouse pointer gradually drifts away from the scrollbar. The longer you drag, the further the mouse gets from the scrollbar, until it completely leaves the scrollbar's interactive area, making it impossible to continue dragging.

**Comment by NeverMorewd** [comment]
After disabling virtualization of the ListBox, both the overlapping issue and scrollbar behavior problem no longer occur. It appears these issues were caused by virtualization.
```
<ListBox.ItemsPanel>
<ItemsPanelTemplate>
	<StackPanel/>
</ItemsPanelTemplate>
</ListBox.ItemsPanel>
```

**Comment by timunie** [comment]
Yeah virtualization with different item sizes is a pain and will always be. If you have a handful of items and can deal without, that's great

**Comment by NeverMorewd** [comment]
> Yeah virtualization with different item sizes is a pain and will always be. If you have a handful of items and can deal without, that's great
Do we have a plan to solve the virtualization problem? Also I noticed that the issue mentioned in the other [issue](https://github.com/AvaloniaUI/Avalonia/issues/10367), although it has been closed, I used the code at the time and upgraded to 11.1.4 ,the virtualization problem still persists.

**Comment by NeverMorewd** [comment]
> Yeah virtualization with different item sizes is a pain and will always be. If you have a handful of items and can deal without, that's great

Do we have a plan to solve the virtualization problem? Also I noticed that the issue mentioned in the other [issue](https://github.com/AvaloniaUI/Avalonia/issues/10367), although it has been closed, I used the code at the time and upgraded to 11.1.4 ,the virtualization problem still persists.

**Comment by timunie** [comment]
@NeverMorewd can you please try to use `Dispather.UiThread.Post(() => { selecting.SelectedItem = whatever; })` in order to give the control enough time to load?

**Comment by timunie** [comment]
[Reproduction.zip](https://github.com/user-attachments/files/17541614/Reproduction.zip)

@grokys fyi attached is a minimal sample of the other linked issue, updated to 11.2.0-rc2. 

Issue is only visible if you:
1. expand a random item 
2. Select this item 
3. Scroll upwards a little bit
4. Notice the selected item is there as a ghost item

@NeverMorewd Timeframe to solve? --> No priority from our side currently unless it becomes a customer issue. If you are a paying customer, let me know. If you want to digg into Avalonia source and file a PR, that is also an option.

**Comment by NeverMorewd** [comment]
> @NeverMorewd can you please try to use `Dispather.UiThread.Post(() => { selecting.SelectedItem = whatever; })` in order to give the control enough time to load?
Thank you. After some test, I found the following code can make sense.
`Dispatcher.UIThread.RunJobs(DispatcherPriority.Loaded);`

**Comment by NeverMorewd** [comment]
> @NeverMorewd can you please try to use `Dispather.UiThread.Post(() => { selecting.SelectedItem = whatever; })` in order to give the control enough time to load?

After some test, the following code appears to make sense
`Dispatcher.UIThread.RunJobs(DispatcherPriority.Loaded);`
